javascript/ql/src/semmle/javascript/dataflow/internal/FlowSteps.qll

```diff
@@ -115,6 +115,35 @@ predicate callStep(DataFlow::Node pred, DataFlow::Node succ) {
     argumentPassing(_, pred, _, parm) and
     succ = DataFlow::parameterNode(parm)
   )
+  or
+  constructorReceiverToMethodStep(pred, succ)
+}
+
+/**
+ * Holds if `pred` is the receiver of a class constructor and `succ` is the receiver of one
+ * of the methods of that class.
+ */
+predicate constructorReceiverToMethodStep(DataFlow::Node pred, DataFlow::Node succ) {
+  exists (ClassDefinition class_ |
+    pred = DataFlow::thisNode(class_.getConstructor().getBody()) and
+    succ = DataFlow::thisNode(class_.getAMethod().getBody())
```

Comment:
A call to `MemberDeclaration::isStatic` is required here to avoid flow between static and non-static members.

Reply:
Should be handled by `ClassNode`.